ui/src/app/pages/workspace/workspace-about.spec.tsx

```diff
@@ -325,4 +327,69 @@ describe('WorkspaceAbout', () => {
       workspace.namespace
     );
   });
+
+  it('renders Community Workspace section when enabled', () => {
+    serverConfigStore.set({
+      config: { ...defaultServerConfig, enablePublishedWorkspacesViaDb: true },
+    });
+    component();
+    expect(screen.getByText('Community Workspace')).toBeInTheDocument();
+  });
+
+  it('does not render Community Workspace section when enablePublishedWorkspacesViaDb disabled', () => {
+    serverConfigStore.set({
+      config: { ...defaultServerConfig, enablePublishedWorkspacesViaDb: false },
+    });
+    component();
+    expect(screen.queryByText('Community Workspace')).not.toBeInTheDocument();
+  });
+
+  it('Publish button is enabled for workspace owner', () => {
+    serverConfigStore.set({
+      config: { ...defaultServerConfig, enablePublishedWorkspacesViaDb: true },
+    });
+    component();
+    const publishButton = screen.getByText('Publish');
+    expectButtonElementEnabled(publishButton);
+  });
+
+  it('Publish button is disabled for non workspace owner', () => {
+    serverConfigStore.set({
+      config: { ...defaultServerConfig, enablePublishedWorkspacesViaDb: true },
+    });
+    const nonOwnerWorkspace = {
+      ...workspaceStubs[0],
+      accessLevel: WorkspaceAccessLevel.WRITER,
+    };
+    currentWorkspaceStore.next(nonOwnerWorkspace);
+    component();
+    const publishButton = screen.getByText('Publish');
+    expectButtonElementDisabled(publishButton);
+  });
+
+  it('Publish button is disabled for workspace owner if workspace is already published', () => {
+    serverConfigStore.set({
+      config: { ...defaultServerConfig, enablePublishedWorkspacesViaDb: true },
+    });
+    const publishedWorkspace = {
+      ...workspaceStubs[0],
+      featuredCategory: FeaturedWorkspaceCategory.COMMUNITY,
+      accessLevel: WorkspaceAccessLevel.OWNER,
+    };
+    currentWorkspaceStore.next(publishedWorkspace);
+    component();
+    const publishButton = screen.getByText('Publish');
+    expectButtonElementDisabled(publishButton);
+  });
+
+  it('Publish button click sets showPublishConsentModal to true', () => {
+    serverConfigStore.set({
+      config: { ...defaultServerConfig, enablePublishedWorkspacesViaDb: true },
+    });
+    component();
+    fireEvent.click(screen.getByText('Publish'));
```

Comment:
We tend to use `user.<action>`, because it better represents all of the given actions that might get triggered when a user interacts with a component. There are a few exceptions, but this page explains it fairly well:
https://testing-library.com/docs/user-event/intro/#differences-from-fireevent